commands/azureCommands/acr-build-logs.ts

```diff
@@ -0,0 +1,381 @@
+        <meta charset="UTF-8">
+        <meta http-equiv="Content-Security-Policy" content="frame-src vscode-resource:; img-src vscode-resource: https:; script-src vscode-resource:; style-src vscode-resource:;">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Logs</title>
+    </head>
+
+    <body>
+        <div id = "header">
+            <table id="headerTable">
+                <th class = 'widthControl'>Build Name </th>
+                <th class = 'widthControl'>BuildTask </th>
+                <th class = 'widthControl'>Status </th>
+                <th class = 'widthControl'>Start Time </th>
+                <th class = 'widthControl'>Finish Time </th>
+                <th class = 'widthControl'>Platform </th>
+                <td></td>
+            </table>
+        </div>
+        <div id = 'core'>
+        </div>
+        <div class = 'loadMoreBtn'>
+            <button id= "loadBtn" class="viewLog">Load More Logs</button>
+        </div>
+        <script src= "${scriptFile}"></script>
+    </body>
+`;
+}
+/** Setup communication with the webview sorting out received mesages from its javascript file */
+function setupCommunication(panel: vscode.WebviewPanel, logData: LogData): void {
+    panel.webview.onDidReceiveMessage(async (message) => {
+        if (message.logRequest) {
+            const itemNumber: number = +message.logRequest.id;
+            logData.getLink(itemNumber).then((url) => {
+                if (url !== 'requesting') {
+                    openLog(url, logData.logs[itemNumber].buildId);
+                }
+            })
+        } else if (message.loadMore) {
+            try {
+                await logData.loadMoreLogs();
+            } catch (error) {
+                vscode.window.showErrorMessage(error.message);
+            }
+
+            addLogsToWebView(panel, logData, logData.logs.length);
+        }
+    });
+}
+
+//# LOGS
+/** Passes the log text into a document provider */
+function createLogView(text: string, title: string): void {
+    const scheme = 'purejs';
+    let uri: vscode.Uri;
+    try {
+        let query = JSON.stringify({ 'log': makeBase64(text) });
+        uri = vscode.Uri.parse(`${scheme}://authority/${title}?${query}#idk`);
+    } catch (error) {
+        console.log(error);
+    }
+
+    vscode.commands.executeCommand('vscode.previewHtml', uri, undefined, title).then(_ => { }, _ => {
+        vscode.window.showErrorMessage('Cant open!');
+    });
+}
+/** Loads log text from remote url using azure blobservices */
+function openLog(url: string, title: string): void {
+    let blobInfo = getBlobInfo(url);
+    let blob: BlobService;
+    try {
+        blob = createBlobServiceWithSas(blobInfo.host, blobInfo.sasToken);
+    } catch (error) {
+        console.log(error);
+    }
+    try {
+        blob.getBlobToText(blobInfo.containerName, blobInfo.blobName, async (error, text, result, response) => {
```

Comment:
I should be able to but one can use the blobservices to stream the logs so I initially attempted to pursue that option, I am gonna re look into that given the new console you were showing me and potential for ongoing logs, otherwise it makes more sense to just download the content as you noted.